pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_base_instrument.py

```diff
@@ -0,0 +1,227 @@
+            elif par[1] == 'bool':
+                self.add_parameter(
+                    parname,
+                    set_cmd=self._gen_set_func(self._dev.seti, parfunc),
+                    get_cmd=self._gen_get_func(self._dev.geti, parfunc),
+                    vals=vals.Ints(int(par[2]), int(par[3])))
+            else:
+                print("parameter {} type {} from from"
+                      " s_node_pars not recognized".format(parname, par[1]))
+
+    def add_d_node_pars(self, filename: str):
+        f = open(filename).read()
+        d_node_pars = json.loads(f)
+        for parameter in d_node_pars:
+            parname = parameter[0].replace("/", "_")
+            parfunc = "/"+self._devname+"/"+parameter[0]
+            if parameter[1] == 'float':
+                self.add_parameter(
+                    parname,
+                    get_cmd=self._gen_get_func(self._dev.getd, parfunc))
+            elif parameter[1] == 'vector_g':
+                self.add_parameter(
+                    parname,
+                    get_cmd=self._gen_get_func(self._dev.getv, parfunc))
+            elif parameter[1] == 'vector_s':
+                self.add_parameter(
+                    parname,
+                    set_cmd=self._gen_set_func(self._dev.setv, parfunc),
+                    vals=vals.Anything())
+            elif parameter[1] == 'vector_gs':
+                self.add_parameter(
+                    parname,
+                    set_cmd=self._gen_set_func(self._dev.setv, parfunc),
+                    get_cmd=self._gen_get_func(self._dev.getv, parfunc),
+                    vals=vals.Anything())
+            else:
+                print("parameter {} type {} from d_node_pars"
+                      " not recognized".format(parname, parameter[1]))
+
+    def _gen_set_func(self, dev_set_type, cmd_str):
+        def set_func(val):
+            dev_set_type(cmd_str, val)
+            return dev_set_type(cmd_str, value=val)
+        return set_func
+
+    def _gen_get_func(self, dev_get_type, ch):
+        def get_func():
+            return dev_get_type(ch)
+        return get_func
+
+    def create_parameter_files(self):
+        '''
+        this functions retrieves all possible settable and gettable parameters
+        from the device.
+        Additionally, it gets all minimum and maximum values for the
+        parameters by trial and error to determine the validator bounds.
+
+        Files are written to the zi_parameter_files folder using the names
+        "{par_type}_node_pars_{dev_type}.json"
+
+        '''
+        # set the file names to write to
+        dev_type = self._dev.daq.getByte(
+            '/{}/features/devtype'.format(self._devname))
+        # Watch out this writes the file to the directory from which you
+        # call this command
+        s_file_name = 's_node_pars_{}.json'.format(dev_type)
+        d_file_name = 'd_node_pars_{}.json'.format(dev_type)
```

Comment:
As the comment says, the files will not be written to the zi_parameter_files folder as is specified by the docstring. Should be changed similar to the current UHFQC driver.
```
path = os.path.abspath(__file__)
dir_path = os.path.dirname(path)
self._s_file_name = os.path.join(
    dir_path, 'zi_parameter_files', 's_node_pars_{}.json'.format(dev_type))
self._d_file_name = os.path.join(
    dir_path, 'zi_parameter_files', 'd_node_pars_{}.json'.format(dev_type))
```